### DISCUSSION TITLE
请教下 想要wireshark抓包查看下协议交互过程，如何进行解密？

### DISCUSSION THREAD

**Discussion by dead4ead** [alibaba/xquic#293] [Q&A] [answered]
test_server 和 test_client 交互，使用的openssl生成的server.key / server.crt
如何生成 (Pre)-Master-Secret log file

**Comment by MPK1** [comment, marked as answer]
Hi @dead4ead,
you have to use `-k` to export the SSL key log file.
You can then Go to Wireshark->Preferences->Protocols->TLS and select the key log file.
Wireshark will then decrypt all packets.
